daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/validation/ValidationSpec.scala

```diff
@@ -0,0 +1,651 @@
+  }
+  private val tweakLookupVersion = Tweak.single[Node] {
+    case nl: Node.NodeLookupByKey[_] => Some(nl.copy(version = changeVersion(nl.version)))
+    case _ => None
+  }
+
+  private val sigLookupTweaks =
+    Map(
+      "tweakLookupTemplateId" -> tweakLookupTemplateId,
+      "tweakLookupKey(None)" -> tweakLookupKey,
+      "tweakLookupResult" -> tweakLookupResult,
+      "tweakLookupVersion" -> tweakLookupVersion,
+    )
+
+  private val insigLookupTweaks =
+    Map(
+      "tweakLookupOptLocation" -> tweakLookupOptLocation,
+      "tweakExerciseKey(None)" -> tweakExerciseKey(tweakOptKeyMaintainersNone),
+    )
+
+  //--[Exercise node tweaks]--
+
+  private val tweakExerciseTargetCoid = Tweak.single[Node] {
+    case ne: Node.NodeExercises[_, _] => Some(ne.copy(targetCoid = changeContractId(ne.targetCoid)))
+    case _ => None
+  }
+  private val tweakExerciseTemplateId = Tweak.single[Node] {
+    case ne: Node.NodeExercises[_, _] => Some(ne.copy(templateId = changeTemplateId(ne.templateId)))
+    case _ => None
+  }
+  private val tweakExerciseChoiceId = Tweak.single[Node] {
+    case ne: Node.NodeExercises[_, _] => Some(ne.copy(choiceId = changeChoiceId(ne.choiceId)))
+    case _ => None
+  }
+  private val tweakExerciseOptLocation = Tweak[Node] {
+    case ne: Node.NodeExercises[_, _] =>
+      tweakOptLocation.run(ne.optLocation).map { x => ne.copy(optLocation = x) }
+    case _ => Nil
+  }
+  private val tweakExerciseConsuming = Tweak.single[Node] {
+    case ne: Node.NodeExercises[_, _] => Some(ne.copy(consuming = changeBoolean(ne.consuming)))
+    case _ => None
+  }
+  private val tweakExerciseActingParties = Tweak[Node] {
+    case ne: Node.NodeExercises[_, _] =>
+      tweakPartySet.run(ne.actingParties).map { x => ne.copy(actingParties = x) }
+    case _ => Nil
+  }
+  private val tweakExerciseChosenValue = Tweak.single[Node] {
+    case ne: Node.NodeExercises[_, _] => Some(ne.copy(chosenValue = changeValue(ne.chosenValue)))
+    case _ => None
+  }
+  private val tweakExerciseStakeholders = Tweak[Node] {
+    case ne: Node.NodeExercises[_, _] =>
+      tweakPartySet.run(ne.stakeholders).map { x => ne.copy(stakeholders = x) }
+    case _ => Nil
+  }
+  private val tweakExerciseSignatories = Tweak[Node] {
+    case ne: Node.NodeExercises[_, _] =>
+      tweakPartySet.run(ne.signatories).map { x => ne.copy(signatories = x) }
+    case _ => Nil
+  }
+  private val tweakExerciseChoiceObservers = Tweak[Node] {
+    case ne: Node.NodeExercises[_, _] =>
+      tweakPartySet.run(ne.choiceObservers).map { x => ne.copy(choiceObservers = x) }
+    case _ => Nil
+  }
+  private val tweakExerciseExerciseResultSome = Tweak[Node] { //sig
+    case ne: Node.NodeExercises[_, _] =>
+      ne.exerciseResult match {
+        case None => Nil
+        case Some(v) =>
+          List(
+            ne.copy(exerciseResult = Some(changeValue(v))),
+            ne.copy(exerciseResult = None),
+          )
+      }
+    case _ => Nil
+  }
```

Comment:
With the `PartialFunction` change for `Tweak.apply`

```suggestion
  private val tweakExerciseExerciseResultSome = Tweak[Node] { //sig
    case ne: Node.NodeExercises[_, _] if ne.exerciseResult.isDefined =>
          List(
            ne.copy(exerciseResult = ne.exerciseResult.map(changeValue)),
            ne.copy(exerciseResult = None),
          )
      }
  }
```